File: solutions/go/05/grundy/main.go
package main

import "fmt"

func nimWinner(piles []int) string {
	x := 0
	for _, p := range piles {
		x ^= p
	}
	if x != 0 {
		return "first"
	}
	return "second"
}

func main() {
	fmt.Println(nimWinner([]int{1, 2, 3}))
	fmt.Println(nimWinner([]int{2, 2}))
}

File: solutions/python/05/grundy/main.py
def nim_winner(piles: list) -> str:
    x = 0
    for p in piles:
        x ^= p
    return "first" if x else "second"


def mex(s: set) -> int:
    m = 0
    while m in s:
        m += 1
    return m


def main() -> None:
    print(nim_winner([1, 2, 3]))
    print(nim_winner([2, 2]))


if __name__ == "__main__":
    main()
